### PR TITLE
changes to clean up ansible syntax and use yaml in a cleaner way

### PR DESCRIPTION
**Pull request by thalunil**
i fixed all of the templates, except 10zone.conf.j2 - this is not done yet.
domain.iteritems() is no longer available

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -3,12 +3,26 @@ unbound_anchorfile: "/var/lib/unbound/root.key"
...
 unbound_logfile: "/var/log/unbound.log"
 unbound_configuration:
-    - verbosity: 1
-    - do-ip4: "yes"
-    - do-ip6: "no"
-    - num-threads: 1
-    - pidfile: "/var/run/unbound.pid"
-    - logfile: "{{unbound_logfile}}"
+  - name: verbosity
+    value: "1"
```
Comment (jdauphant) [inline defaults/main.yml:7]: @thalunil this syntax is little more complicated, it's not possible to keep the old one ?
(we lost the compatility also)

File: defaults/main.yml [hunk 2]
```diff
@@ -3,12 +3,26 @@ unbound_anchorfile: "/var/lib/unbound/root.key"
...
+  - name: logfile
+    value: "{{unbound_logfile}}"
+  - name: use-syslog
+    value: "no"
+  - name: log-queries
+    value: "no"
+  - name: do-not-query-localhost
+    value: "yes" # yes is default, no for using a forwarder running on localhost (e.g. dnscrypt-proxy)
+  - name: do-daemonize
+    value: "no" # for systemd systems, default is yes
```
Comment (jdauphant) [inline defaults/main.yml:25]: This last value cause some trouble to the CI

File: tasks/main.yml [hunk 3]
```diff
@@ -48,7 +55,7 @@
   template:
      src=unbound.conf.j2
      dest=/etc/unbound/unbound.conf
-     validate="/usr/sbin/unbound-checkconf %s"
+     #validate="/usr/sbin/unbound-checkconf %s"
```
Comment (jdauphant) [inline tasks/main.yml:58]: Have you find some trouble with the validation ?

Comment (JoeButler99) [reply tasks/main.yml:58]: I think I found out why this was commented - "fatal error: config file /home/ec2-user/.ansible/tmp/ansible-tmp-1543938114.2980123-1210814837092/source is not inside chroot /etc/unbound"
I can't think of an way to fix this as Ansible is checking against the tmp location

File: defaults/main.yml (hunk 1 above)
Comment (rogierm) [reply defaults/main.yml:7]: It is possible by changing the vars and the jinja2 template: 

vars:
    unbound_configuration:
      verbosity: "2"
      do-ip4: "yes"
      do-ip6: "yes"
      num-threads: "1"

01general.conf.j2:

server:
{% for key,value in unbound_configuration.items() %}
    {{ key }}: {{ value }}
{% endfor %}

Comment (rogierm) [reply defaults/main.yml:7]: in 10zone.conf.j2 change iteritems() to items(). I'll create a pull request if you want and if this issue is still open.

Comment (jdauphant) [reply defaults/main.yml:7]: @rogierm thanks, @thalunil looks inactive on his PR
I will be able to merge if you can remove "do-daemonize" https://github.com/jdauphant/ansible-role-unbound/pull/9/files#r210898409 and restore the check https://github.com/jdauphant/ansible-role-unbound/pull/9/files#r210898567

**Comment by jdauphant** [comment]
@thalunil I close this PR, @nerdlich have fixed the role for ansible 2.5 compatibility in a way that keep the config compatible, you can update on his work.
Thanks for your help

**Pull request closed without merging**